File: App/BE/src/main/java/com/project/webbanhang/dtos/ProductDTO.java
package com.project.webbanhang.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.*;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

@Data
@Builder
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class ProductDTO {
    @NotBlank(message = "Product name is required")
    @Size(min = 3, max = 100, message = "Product name must be between 3 and 100 characters")
    private String name;

    @Min(value = 0, message = "Price must be greater than 0")
    @Max(value = 1000000009, message = "Price must be less than 1,000,000,000")
    private float price;
    private String thumbnail;
    private String description;

    @JsonProperty("category_id")
    private Long categoryId;

    private List<MultipartFile> files;
}

File: App/angular/src/app/dtos/product.dto.ts
export interface ProductDTO {
  id: number;
  name: string;
  price: number;
  thumbnail: string;
  description: string;
  created_at: string;
  updated_at: string;
  category_id: number;
  images: string[];
}

export interface ProductResponse {
  products: ProductDTO[];
  totalPage: number;
  totalItem: number;
}
